spring-cloud-dataflow-completion/src/main/java/org/springframework/cloud/dataflow/completion/ValueHintProvider.java

```diff
@@ -22,19 +22,34 @@
 import org.springframework.boot.configurationmetadata.ValueHint;
 
 /**
- * Pluggable interface to provide value hints that can be discovered for some properties.
+ * Interface to provide value hints that can be discovered for properties.
  *
  * @author Eric Bottard
  */
 public interface ValueHintProvider {
 
-	List<ValueHint> guessValueHints(ConfigurationMetadataProperty property, ClassLoader classLoader);
+	/**
+	 * For a given property, return a list of {@link ValueHint} that may apply.
+	 *
+	 * @param property     property for which to generate value hints
+	 * @param classLoader  class loader for the artifact/module that this
```

Comment:
Tried to explain what this `ClassLoader` is for; hopefully this makes sense and the terminology is correct in this context.